### PR TITLE
Scale and Update Parall with revert partition

### PR DESCRIPTION
**Pull request by ColdsteelRail**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y  #174 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/controllers/collaset/synccontrol/update.go [hunk 1]
```diff
@@ -207,7 +213,12 @@ func decidePodToUpdateByLabel(_ *appsv1alpha1.CollaSet, podInfos []*PodUpdateInf
 	return podToUpdate
 }
 
-func decidePodToUpdateByPartition(cls *appsv1alpha1.CollaSet, podInfos []*PodUpdateInfo) (podToUpdate []*PodUpdateInfo) {
+func decidePodToUpdateByPartition(
```
Comment (wu8685) [inline pkg/controllers/collaset/synccontrol/update.go:216]: I think we should treat the NotCreated Pod as placeholder. These Pod placeholders should participate into order operation. Then it is able to get rid of `decideNeedUpdateButNotCreate`.

File: pkg/controllers/collaset/synccontrol/sync_control.go [hunk 2]
```diff
@@ -511,6 +546,14 @@ func (r *RealSyncControl) Scale(
 				// add an expectation for this pod creation, before next reconciling
 				return collasetutils.ActiveExpectations.ExpectCreate(cls, expectations.Pod, pod.Name)
 			})
+			if err != nil || succCount > 0 {
```
[the remark this answers is not in the document]
Comment (ColdsteelRail) [reply pkg/controllers/collaset/synccontrol/sync_control.go]: fixed with: https://github.com/KusionStack/operating/pull/219/commits/8bb068dfabd63731fd6c7fa0b882df415d47d1e6

File: pkg/controllers/collaset/synccontrol/update.go (hunk 1 above)
Comment (ColdsteelRail) [reply pkg/controllers/collaset/synccontrol/update.go:230]: fixed by constructing placeholder podwrappers for ownedIDs

File: pkg/controllers/collaset/utils/pod.go [hunk 3]
```diff
@@ -40,6 +40,7 @@ type PodWrapper struct {
 	ID            int
 	ContextDetail *appsv1alpha1.ContextDetail
 
+	OnlyPlaceholder bool
```
Comment (wu8685) [inline pkg/controllers/collaset/utils/pod.go:43]: Just `PlaceHolder` is ok.

**Review by wu8685 — APPROVED** [review]
lgtm

**Pull request merged**